### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Shevtsooov**
[DEMO LINK](https://Shevtsooov.github.io/react_static-list-of-posts/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -2,108 +2,31 @@ import React from 'react';
...
+import { CompletePost } from './types/CompletePost';
+import { PostList } from './components/PostList';
+
+const completePosts: CompletePost[] = postsFromServer.map(post => {
+  const user = usersFromServer.find(
+    serverUser => serverUser.id === post.userId,
+  );
+  const comments = commentsFromServer.filter(comment => (
+    comment.postId === post.id
+  ));
```
Comment (silchyn) [inline src/App.tsx:18]: could be separate functions

File: src/components/CommentInfo/CommentInfo.tsx [hunk 2]
```diff
@@ -1,5 +1,31 @@
 import React from 'react';
+import { Comments } from '../../types/Comments';
+import './CommentInfo.scss';
+
+interface Props {
+  comment: Comments,
+}
+
+export const CommentInfo: React.FC<Props> = ({ comment }) => (
```
Comment (silchyn) [inline src/components/CommentInfo/CommentInfo.tsx:9]: destructure `comment`

File: src/components/PostInfo/PostInfo.tsx [hunk 3]
```diff
@@ -1,5 +1,42 @@
...
+import './PostInfo.scss';
+import { CompletePost } from '../../types/CompletePost';
+import { CommentList } from '../CommentList';
+import { UserInfo } from '../UserInfo/UserInfo';
+
+type PostInfoProps = {
+  post: CompletePost,
+};
+
+export const PostInfo: React.FC<PostInfoProps> = ({ post }) => (
```
Comment (silchyn) [inline src/components/PostInfo/PostInfo.tsx:12]: destructure `post`

File: src/components/PostInfo/PostInfo.tsx [hunk 4]
```diff
@@ -1,5 +1,42 @@
...
+    {post?.comments.length
+      ? <CommentList comments={post.comments} />
+      : (
+        <b
+          className="PostInfo__NoCommentsMessage"
+          data-cy="NoCommentsMessage"
+        >
+          No comments yet
+        </b>
+      )}
```
Comment (silchyn) [inline src/components/PostInfo/PostInfo.tsx:40]:
```suggestion
      {post.user && (
        <p>
          {' Posted by '}
          <UserInfo user={post.user} />
        </p>
      )}
    </div>

    <p className="PostInfo__body">
      {post.body}
    </p>

    <hr />
    
    {post?.comments.length
      ? <CommentList comments={post.comments} />
      : (
        <b
          className="PostInfo__NoCommentsMessage"
          data-cy="NoCommentsMessage"
        >
          No comments yet
        </b>
      )}
```

File: src/components/PostList/PostList.tsx [hunk 5]
```diff
@@ -1,5 +1,16 @@
...
 
-export const PostList: React.FC = () => (
-  <>Put the list here</>
+interface Props {
+  posts: CompletePost[],
+}
+
+export const PostList: React.FC<Props> = ({ posts }) => (
+  <div className="PostList">
+
```
Comment (silchyn) [inline src/components/PostList/PostList.tsx:11]:
```suggestion
```

File: src/components/UserInfo/UserInfo.tsx [hunk 6]
```diff
@@ -1,5 +1,16 @@
...
+import './UserInfo.scss';
+import { User } from '../../types/User';
 
-export const UserInfo: React.FC = () => (
-  <>Put the user here</>
+type UserInfoProps = {
+  user: User,
+};
+
+export const UserInfo: React.FC<UserInfoProps> = ({ user }) => (
```
Comment (silchyn) [inline src/components/UserInfo/UserInfo.tsx:9]: destructure `user`

**Review by silchyn — CHANGES_REQUESTED** [review]
Pay attention to your code style

**Comment by Shevtsooov** [comment]
@silchyn Hey, thanks for the review!
The code style was fixed, please check.

**Review by silchyn — APPROVED** [review]

**Pull request closed without merging**